nservicebus/transactional-session/index.md

```diff
@@ -0,0 +1,155 @@
+
+* [Azure Table](/persistence/azure-table)
+* [CosmosDB](/persistence/cosmosdb)
+* [SQL](/persistence/sql)
+* [NHibernate](/persistence/nhibernate)
+* [RavenDB](/persistence/ravendb)
+* [MongoDB](/persistence/mongodb)
+
+## Transaction consistency
+
+To guarantee atomic consitency across database and message operations, the transactional session requires the [Outbox](/nservicebus/outbox) to be enabled.
+
+With the Outbox disabled, database and message operations are not executed until the session is committed. All database operations shre the same database transaction but message operations are not guaranteed to be atomic with the database changes. This might lead to phantom records or ghost messages when in case of a failure during the commit phase.doc
+
+## How it works
+
+The transactional session feature guarantees that all outgoing message operations are eventually consistent with the data operations.
+
+Returning to the earlier example of a message handler which creates a `User` and then publishes a `UserCreated` event, the following process occurs. Details are described following the diagram.
+
+```mermaid
+sequenceDiagram
+    actor User
+    autonumber
+    Note over User: Phase 1
+    activate User
+    User->>TransactionalSession: Open()
+    activate TransactionalSession
+    TransactionalSession->>PendingTransportOperations: new()
+    TransactionalSession->>Storage: BeginTransaction()
+    activate Storage
+    Storage-->>TransactionalSession: transaction
+    deactivate Storage
+    deactivate TransactionalSession
+
+    User->>TransactionalSession: Publish<UserCreated>()
+    activate TransactionalSession
+    TransactionalSession->>PendingTransportOperations: Add()
+    deactivate TransactionalSession
+    activate TransactionalSession
+    deactivate TransactionalSession
+    User->>Storage: Store(user)
+
+    User->>TransactionalSession: Commit()
+    activate TransactionalSession
+    TransactionalSession->>Transport: Dispatch(controlMessage)
+    TransactionalSession->>PendingTransportOperations: ConvertToOutboxOperations
+    activate PendingTransportOperations
+    PendingTransportOperations-->>TransactionalSession: outboxRecord
+    deactivate PendingTransportOperations
+    TransactionalSession->>Storage: Store(outboxRecord)
+    TransactionalSession->>Storage: Complete()
+    deactivate TransactionalSession
+    deactivate User
+
+    Note over ReceivePipeline: Phase 2
+    ReceivePipeline->>Transport: Read()
+    activate ReceivePipeline
+    ReceivePipeline->>Storage: GetOutboxRecord()
+    activate Storage
+    Storage-->>ReceivePipeline: outboxRecord
+    deactivate Storage
+    ReceivePipeline->>Transport: Dispatch(outboxRecord.Operations)
+    ReceivePipeline->>Storage: SetAsDispatched()
+    ReceivePipeline-->>Transport: Ack()
+    deactivate ReceivePipeline
+```
+
+There is no single transaction that spans all the operations. The operations occur in two separate phases:
```

Comment:
Does this scare off people because we talk about transactionality all the time? Should we maybe slightly twist this to something like

> Internally the transactional session doesn't use a single transaction that spans all the operations. The transactional session ackknowledgement occurs in two seperate phases: